runtime/server/auth/interceptors.go

```diff
@@ -95,6 +97,22 @@ func HTTPMiddleware(aud *Audience, next http.Handler) http.Handler {
 	})
 }
 
+// PSQLAuthHandler handles authentication for a PostgreSQL wire-compatible endpoint
+func PSQLAuthHandler(aud *Audience) func(ctx context.Context, username, password string) (context.Context, bool, error) {
+	return func(ctx context.Context, username, password string) (context.Context, bool, error) {
+		var err error
+		password, err = url.QueryUnescape(password)
+		if err != nil {
+			return nil, false, err
+		}
+		ctxWithClaim, err := parseClaims(ctx, aud, password)
+		if err != nil {
+			return ctx, false, fmt.Errorf("authentication failed: %w", err)
+		}
```

Comment:
To avoid the `bearer` part here, you probably need to split `parseClaims` into `parseClaims(ctx, aud, jwt)` and `parseClaimsFromBearer(ctx, aud, authorizationHeader)`.